class.jetpack.php

```diff
@@ -4736,6 +4739,26 @@ function authenticate_jetpack( $user, $username, $password ) {
 		return new WP_User( $token_details['user_id'] );
 	}
 
+	// Authenticates requests from Jetpack server to WP API endpoints.
+	// Uses the existing XMLRPC oAuth implementation.
+	function wp_json_authenticate( $error ) {
+		if ( isset( $_GET['token'] ) ) {
+			$verified = $this->verify_xml_rpc_signature();
+
+			if ( is_wp_error( $verified ) ) {
+				return $verified;
+			}
+
+			if ( isset( $verified['type'] ) && isset( $verified['user_id'] ) && 'user' === $verified['type']  ) {
+				wp_set_current_user( $verified['user_id'] );
+				return true;
```

Comment:
More correct to return `null` here - that is the default value passed to the filter, and `rest_authentication_errors -> true` reads a bit strangely.